onnxruntime/python/tools/transformers/convert_to_onnx.py

```diff
@@ -181,15 +196,25 @@ def main():
 
     session = create_onnxruntime_session(output_path, args.use_gpu, enable_all_optimization=True, verbose=args.verbose)
     if session is not None:
-        Gpt2Helper.test_parity(session,
-                               model,
-                               device,
-                               args.precision == Precision.FLOAT16,
-                               rtol=args.tolerance,
-                               atol=args.tolerance,
-                               model_class=args.model_class,
-                               has_position_ids=use_padding,
-                               has_attention_mask=use_padding)
+        if use_beam_search_step:
```

Comment:
Can we add a factory class here that would return an instance of `Gpt2Helper` or `Gpt2BeamSearchHelper` depending on the value of the `use_beam_search_step parameter`? This way we would less code duplication.

For instance:

```
class Gpt2HelperFactory:
    @staticmethod
    def create_helper(helper_type="default"):
        helpers = {
            "default": Gpt2Helper,
            "beam_search_step": Gpt2BeamSearchHelper,
        }
        w, args = helpers[gpt2helper_type]
        if args:
            return w(*args)
        return w()
```

And then call without if-else:L
`Gpt2HelperFactory[helper_type].test_parity(....)`

cc @skdeng

Reply:
Good suggestions. Thanks. I have added Gpt2HelperFactory class. Meanwhile, I have sub-classed Gpt2BeamSearchHelper and Gpt2BeamSearchInput for less code duplication